esmvalcore/preprocessor/_io.py

```diff
@@ -269,3 +281,63 @@ def _write_ncl_metadata(output_dir, metadata):
     write_ncl_settings(info, filename)
 
     return filename
+
+
+def _concatenate_overlapping_cubes(cubes):
+    """Concatenate time-overlapping cubes (two cubes only)."""
+    # we arrange [cube1, cube2] so that cube1.start <= cube2.start
+    if cubes[0].coord('time').points[0] <= cubes[1].coord('time').points[0]:
+        cubes = [cubes[0], cubes[1]]
+    else:
+        cubes = [cubes[1], cubes[0]]
+
+    # get time end points
+    time_1 = cubes[0].coord('time')
+    time_2 = cubes[1].coord('time')
+    data_start_1 = time_1.cell(0).point
+    data_start_2 = time_2.cell(0).point
+    data_end_1 = time_1.cell(-1).point
+    data_end_2 = time_2.cell(-1).point
+
+    # case 1: both cubes start at the same time -> return longer cube
+    if data_start_1 == data_start_2:
+        if data_end_1 < data_end_2:
+            cubes = [cubes[1]]
+        else:
+            cubes = [cubes[0]]
+
+    # case 2: cube1 starts before cube2
+    else:
+        # find time overlap, if any
+        start_overlap = next(
+            (time_1.units.num2date(t) for t in time_1.points
+             if t in time_2.points),
+            None
+        )
+        # case 0: no overlap (new iris implementaion does allow
```

Comment:
This seems to conflict with `# case 2.3: there is no overlap: return original cubes` defined on line 341. Also, I think this check is not needed here, because a discontinuous time coordinate is something that should be found by the cmor check.

Reply:
Yeah that line 341 is not actually a case per-se (need to remove the comment) - it's safer to raise the error if no overlap than to allow the concatenation to happen and then have it found by cmor.check when looking for time discontinuities

Reply:
I'm not sure, because in the usual case no overlap is expected. E.g. if concatenation fails because there are problems related to coordinates other than the time coordinate, you will get this error, which is really confusing..

Reply:
Ermm, that is true, but then again, having no overlap in time is still a valid issue with the cubes that must be reported...

Reply:
Oh actually, yeah, you're right! in a standard case there is no overlap and the lack of it will be reported - we need to remove that error, maybe make it a logger debug or warning?

Reply:
I think it's enough to simply remove the check and exception here